admin/services/data-access/group.data.service.js

```diff
@@ -20,17 +25,76 @@ groupDataService.getGroups = async function (query) {
     .exec()
 }
 
+/**
+ * Get active groups (non-soft-deleted).
+ * @returns {Promise<*>}
+ */
+groupDataService.sqlGetGroups = async () => {
+  const sql = `SELECT g.* 
+    FROM ${sqlService.adminSchema}.[group] g 
+    WHERE g.isDeleted=0 
+    ORDER BY name ASC`
+  return sqlService.query(sql)
+}
+
 /**
  * Get group document by _id.
+ * @deprecated use sqlGetGroup
  * @param query
  * @returns {Promise<*>}
  */
 groupDataService.getGroup = async function (query) {
   return Group.findOne(query).lean().exec()
 }
 
+/**
+ * Get group by id.
+ * @param groupId
+ * @returns {Promise<void>}
+ */
+groupDataService.sqlGetGroup = async (groupId) => {
+  const sql = `SELECT g.* 
+    FROM ${sqlService.adminSchema}.[group] g 
+    WHERE g.id=@groupId`
+
+  const params = [
+    {
+      name: 'groupId',
+      value: groupId,
+      type: TYPES.Int
+    }
+  ]
+
+  const result = await sqlService.query(sql, params)
+  return R.head(result)
+}
+
+/**
+ * Get group by name.
+ * @param groupName
+ * @returns {Promise<void>}
+ */
+groupDataService.sqlGetGroupByName = async (groupName) => {
+  const sql = `SELECT g.* 
+    FROM ${sqlService.adminSchema}.[group] g 
+    WHERE isDeleted=0
+    AND g.name=@groupName`
+
+  const params = [
+    {
+      name: 'groupName',
+      value: groupName,
+      type: TYPES.NVarChar
+    }
+  ]
+
+  const result = await sqlService.query(sql, params)
```

Comment:
As it is possible to have multiple groups with the same name in the DB, this could return more than one result.  So - are you sure a db-constraint is not required?